ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -162,8 +188,13 @@ private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameN
       rewrittenCtx.setOperation(Context.Operation.UPDATE);
       rewrittenCtx.addDestNamePrefix(1, Context.DestClausePrefix.UPDATE);
     } else if (deleting()) {
-      rewrittenCtx.setOperation(Context.Operation.DELETE);
-      rewrittenCtx.addDestNamePrefix(1, Context.DestClausePrefix.DELETE);
+      if (isOverwrite) {
+        rewrittenCtx.setOperation(Context.Operation.OTHER);
```

Comment:
why type should be other?

Reply:
This guy throws me out:
```
   * {@link org.apache.hadoop.hive.ql.io.AcidUtils.Operation}.  See usages
   * of {@link #getDestNamePrefix(ASTNode, QB)} and
   * {@link org.apache.hadoop.hive.ql.parse.SemanticAnalyzer#updating(String)} and
   * {@link org.apache.hadoop.hive.ql.parse.SemanticAnalyzer#deleting(String)}.
   */
  public DestClausePrefix getDestNamePrefix(ASTNode curNode, QB queryBlock) {
```
With:
```
 java.lang.IllegalStateException: Expected 'insert overwrite table default.ice01 select *  from default.ice01' to be in sub-query or set operation.
        at org.apache.hadoop.hive.ql.Context.getDestNamePrefix(Context.java:297)

```

Reply:
ok, then why do we need then to set it explicitly
````
rewrittenCtx.addDestNamePrefix(1, Context.DestClausePrefix.INSERT);
````

Reply:
@ayushtkn, not sure if you got my comment right. I was saying why can we simply 
````
rewrittenCtx.setOperation(Context.Operation.OTHER);
````
if that would set 
````
rewrittenCtx.addDestNamePrefix(1, Context.DestClausePrefix.INSERT);
````
in getDestNamePrefix
but, if
````
rewrittenCtx.addDestNamePrefix(1, Context.DestClausePrefix.INSERT);
```` 
is enough that's OK